cdap-kubernetes/src/main/java/io/cdap/cdap/master/environment/k8s/KubeMasterEnvironment.java

```diff
@@ -291,4 +335,61 @@ private PodInfo createPodInfo(Map<String, String> conf) throws IOException, ApiE
   private boolean isCustomVolumePrefix(String name) {
     return CUSTOM_VOLUME_PREFIX.stream().anyMatch(name::startsWith);
   }
+
+  private String getMasterPath() {
+    // Spark master base path
+    String master;
+    // apiClient.getBasePath() returns path similar to https://10.8.0.1:443
+    try {
+      ApiClient apiClient = Config.defaultClient();
+      master = apiClient.getBasePath();
+    } catch (Exception e) {
+      // should not happen
+      throw new RuntimeException("Exception while getting kubernetes master path", e);
+    }
+    return master;
+  }
+
+  private V1PodSpec getPodSpecWithCConf(PodInfo podInfo) {
+    List<V1Volume> volumes = podInfo.getVolumes();
+    List<V1VolumeMount> containerVolumeMounts = podInfo.getContainerVolumeMounts();
+    V1Volume cConfVolume = null;
+    V1VolumeMount cConfVolumeMount = null;
+
+    for (V1VolumeMount containerVolumeMount : containerVolumeMounts) {
+      // Only mount cdap conf to spark driver pod.
+      if (!CDAP_CONF_PATH.equals(containerVolumeMount.getMountPath())) {
+        continue;
+      }
+      for (V1Volume volume : volumes) {
+        if (volume.getName().equals(containerVolumeMount.getName())) {
+          cConfVolume = volume;
+          cConfVolumeMount = containerVolumeMount;
+          break;
+        }
+      }
+    }
+
+    V1PodSpecBuilder v1PodSpecBuilder = new V1PodSpecBuilder();
+
```

Comment:
nit: it seems like this blank new line can be avoided here

Reply:
done